core/src/main/java/org/mule/routing/MessageFilter.java

```diff
@@ -188,4 +190,13 @@ public void dispose()
 
         LifecycleUtils.disposeIfNeeded(filter, LOGGER);
     }
+
+    @Override
+    public void addMessageProcessorPathElements(MessageProcessorPathElement pathElement)
+    {
+        if (!(next instanceof MessageProcessorChain))
```

Comment:
After thinking about this, I realised that this is not the real fix for the issue. Will reopen the PR after I find out the real fix.